File: app/src/main/java/com/example/geom_databinding/recyclerview/RecyclerViewActivity.java
package com.example.geom_databinding.recyclerview;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;
import android.widget.Toast;

import com.example.geom_databinding.R;
import com.example.geom_databinding.databinding.ActivityRecyclerviewBinding;
import com.example.geom_databinding.main.MainActivity;
import com.example.geom_databinding.model.Point;

import java.util.List;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.Observable;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProviders;
import io.reactivex.Flowable;

public class RecyclerViewActivity extends AppCompatActivity {

    private final String TAG = getClass().getSimpleName();

    // ViewModel, DataBinding
    private RecyclerViewModel recyclerViewModel;
    private ActivityRecyclerviewBinding recyclerviewBinding;

    // List
    private List<Point> points;
    private ActionBar actionBar;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);

        setupActionBar(actionBar); // 뒤로가기 버튼 설정
        setupRecyclerViewBindings(); // 데이터바인딩 셋팅(초기화)
        setupDataOnRecyclerView();  // 리싸이클러뷰 셋팅
    }

    private void setupActionBar(ActionBar actionBar){
        if(actionBar == null){
            actionBar = getSupportActionBar();
        }

        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setDisplayShowHomeEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        int id = item.getItemId();

        switch(id){
            case android.R.id.home :
                //Intent intent = new Intent(this, MainActivity.class);
                //startActivity(intent);
                onBackPressed();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public void setupRecyclerViewBindings(){
        recyclerViewModel = ViewModelProviders.of(this).get(RecyclerViewModel.class);
        recyclerViewModel.init();

        // 데이터바인딩
        recyclerviewBinding = DataBindingUtil.setContentView(this, R.layout.activity_recyclerview);
        recyclerviewBinding.setViewmodel(recyclerViewModel);
        recyclerviewBinding.setLifecycleOwner(this);
    }

    private void setupDataOnRecyclerView(){
        Intent intent = getIntent();
        points = (List<Point>) intent.getSerializableExtra("LIST");

        Log.d(TAG, "points.size() : " + points.size());

        if(points.size() > 0){

            // 어답터에 리스트 연결
            recyclerViewModel.setRecyclerViewAdapter(points);
        }

        setupListClick();
    }

    private void setupListClick(){
        recyclerViewModel.getSelected().observe(this, new Observer<Point>() {
            @Override
            public void onChanged(Point point) {
                if(point != null){
                    Toast.makeText(RecyclerViewActivity.this, "position : " + point.getPosition(), Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/geom_databinding/recyclerview/RecyclerViewActivity.kt
package com.example.geom_databinding.recyclerview;

import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.ActionBar
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.geom_databinding.R
import com.example.geom_databinding.base.BaseActivity
import com.example.geom_databinding.databinding.ActivityRecyclerviewBinding
import com.example.geom_databinding.model.Point

class RecyclerViewActivity : BaseActivity() {

    private val TAG = javaClass.name

    // ViewModel, DataBinding
    private lateinit var mViewModel: RecyclerViewModel
    private lateinit var mBinding: ActivityRecyclerviewBinding

    // List
    private lateinit var points: List<Point>
    private var actionBar: ActionBar? = null

    override fun before() {
        setupActionBar() // 뒤로가기 버튼 설정
        setupRecyclerViewBindings() // 데이터바인딩 셋팅(초기화)
        setupDataOnRecyclerView()  // 리싸이클러뷰 셋팅
    }

    override fun setupObserving() {
        mViewModel.posMessageLiveData.observe(this, Observer {
            Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
        })
    }

    override fun after() {
    }

    override fun initToolbar() {
    }

    private fun setupActionBar(){
        if(actionBar == null){
            actionBar = supportActionBar
        }
        actionBar!!.setDisplayHomeAsUpEnabled(true)
        actionBar!!.setDisplayShowHomeEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean{
        when(item.itemId){
            (android.R.id.home)->
                //Intent intent = new Intent(this, MainActivity.class)
                //startActivity(intent)
                onBackPressed()
            else -> Exception("???")
        }

        return super.onOptionsItemSelected(item)
    }

    private fun setupRecyclerViewBindings(){
        mViewModel = ViewModelProviders.of(this).get(RecyclerViewModel::class.java)
        mViewModel.init()

        // 데이터바인딩
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_recyclerview)
        mBinding.viewmodel = mViewModel
        mBinding.lifecycleOwner = this
    }

    private fun setupDataOnRecyclerView(){
        val intent = intent
        points = intent.getSerializableExtra("LIST") as List<Point>

        if(!points.isNullOrEmpty()){

            // 어답터에 리스트 연결
            mViewModel.setRecyclerViewAdapter(points)
        }
    }
}
